tasks/broad/RNAWithUMIsTasks.wdl

```diff
@@ -0,0 +1,576 @@
+    meta {
+        description: "Verify that the pipeline input is either a ubam or pair of fastqs with additional info"
+    }
+
+    input {
+        File? bam
+        File? r1_fastq
+        File? r2_fastq
+
+        # fastq specific field
+        String? platform
+        String? library_name
+        String? platform_unit
+        String? read_group_name
+        String? sequencing_center = "BI"
+
+        String docker = "python:3.7.2"
+        Int cpu = 1
+        Int memory_mb = 1000
+        Int disk_size_gb = 1
+    }
+
+    command <<<
+        set -e
+        python3 <<CODE
+
+        fastq_flag = 0
+        bam = "~{bam}"
+        r1_fastq = "~{r1_fastq}"
+        r2_fastq = "~{r2_fastq}"
+        platform = "~{platform}"
+        library_name = "~{library_name}"
+        platform_unit = "~{platform_unit}"
+        read_group_name = "~{read_group_name}"
+        sequencing_center = "~{sequencing_center}"
+
+        if bam and not r1_fastq and not r2_fastq:
+            pass
+        elif r1_fastq and r2_fastq and not bam:
+            if platform and library_name and platform_unit and read_group_name and sequencing_center:
+                fastq_flag += 1
+        else:
+            raise ValueError("Invalid Input. Input must be either ubam or pair of fastqs with supplemental data")
+
+        with open("output.txt", "w") as f:
+            if fastq_flag == 1:
+                f.write("true")
+            # Remaining case is that only bam is defined
+            else: 
+                f.write("false")
+
+        CODE
+    >>>
+
+    runtime {
+        docker: docker
+        cpu: cpu
+        memory: "${memory_mb} MiB"
+        disks: "local-disk ${disk_size_gb} HDD"
+    }
+
+    output {
+        Boolean fastq_run = read_boolean("output.txt")
+    }
+}
+
+task ExtractUMIs {
+	input {
+		File bam
+		String read1Structure
+		String read2Structure
+
+        String docker = "us.gcr.io/broad-gotc-prod/fgbio:1.0.0-1.4.0-1638817487"
+        Int cpu = 4
+        Int memory_mb = 8000
+        Int disk_size_gb = ceil(2.2 * size(bam, "GB")) + 50
+	}
+
+	command <<<
+		java -jar /usr/gitc/fgbio.jar ExtractUmisFromBam --input ~{bam} \
+			--read-structure ~{read1Structure} \
+			--read-structure ~{read2Structure} \
+			--molecular-index-tags RX \
+			--output extractUMIs.out.bam
+	>>>
+
+	runtime {
+		docker : docker
+		cpu : cpu
+		memory : "${memory_mb} MiB"
+		disks : "local-disk ${disk_size_gb} HDD"
+		preemptible: 0
+	}
+
+	output {
+		File bam_umis_extracted = "extractUMIs.out.bam"
+	}
+}
+
+task STAR {
+	input {
+		File bam
+		File starIndex
+
+        String docker = "us.gcr.io/broad-gotc-prod/neovax-tag-rnaseq:v1"
+        Int cpu = 8
+        Int memory_mb = 64000
+        Int disk_size_gb = ceil(2.2 * size(bam, "GB") + size(starIndex, "GB")) + 250
+	}
+
+	command <<<
+		echo $(date +"[%b %d %H:%M:%S] Extracting STAR index")
+		mkdir star_index
+		tar -xvf ~{starIndex} -C star_index --strip-components=1
+
+		STAR \
```

Comment:
@gbggrant Looking at this a little more...in Optimus for StarAlignment with ubam we are apparently doing the same thing [here](https://github.com/broadinstitute/warp/blob/7a97693dc1cfb4dadb4e723dd2feec9e17fd0b4f/tasks/skylab/StarAlign.wdl#L50)